File: Insert in middle of linked list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
def insert_middle(head, data):
    if head is None:
        return Node(data)

    # Find middle (slow/fast pointer)
    slow, fast = head, head
    while fast.next and fast.next.next:
        slow = slow.next
        fast = fast.next.next

    new_node = Node(data)
    new_node.next = slow.next
    slow.next = new_node
    return head

File: Insert in middle of linked list.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->next = NULL;
    return node;
}
struct Node* insertMiddle(struct Node* head, int data) {
    if (!head) return newNode(data);

    struct Node* slow = head, *fast = head;
    while (fast->next && fast->next->next) {
        slow = slow->next;
        fast = fast->next->next;
    }
    struct Node* node = newNode(data);
    node->next = slow->next;
    slow->next = node;
    return head;
}
